test/e2e/aws_test.go

```diff
@@ -368,8 +368,79 @@ var _ = Describe("functional tests", func() {
 			deleteCluster(setup.namespace, setup.clusterName)
 		})
 	})
+
+	Describe("Delete infra node directly from infra provider", func() {
+		It("Machine referencing deleted infra node should come to failed state", func() {
+			By("Creating a workload cluster with single control plane")
+			makeSingleControlPlaneCluster(namespace, clusterName, awsClusterName, cpAWSMachinePrefix, cpBootstrapConfigPrefix, cpMachinePrefix, instanceType, testTmpDir, false)
+
+			By("Creating the MachineDeployment")
+			createMachineDeployment(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, mdBootstrapConfig, instanceType, initialReplicas, nil, nil)
+
+			By("Deleting node directly from infra cloud")
+			machines, err := getMachinesOfDeployment(namespace, machineDeploymentName)
+			Expect(err).To(BeNil())
+			Expect(len(machines.Items)).Should(BeNumerically(">", 0))
+			terminateEc2Instance(*machines.Items[0].Spec.ProviderID)
+			verifyMachinePhase(namespace, machines.Items[0].Name, clusterv1.MachinePhaseFailed)
+
+			By("Deleting the Cluster")
+			deleteCluster(namespace, clusterName)
+		})
+	})
 })
 
+func verifyMachinePhase(namespace, machineName string, phase clusterv1.MachinePhase) {
+	fmt.Fprintf(GinkgoWriter, "Ensuring machine %s's state is %s ... \n", machineName, string(phase))
+	machine := &clusterv1.Machine{}
+	Eventually(
+		func() bool {
+			if err := kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: machineName}, machine); err != nil {
+				return false
+			}
+			return machine.Status.Phase == string(phase)
+		}, 10*time.Minute, 15*time.Second,
+	).Should(BeTrue())
+}
+
+func terminateEc2Instance(instanceId string) {
```

Comment:
```suggestion
func terminateInstance(instanceId string) {
```

Nit, given that we're in AWS land

Reply:
renamed the function name